pkg/webui/console/containers/notifications/notification-list/index.js

```diff
@@ -0,0 +1,162 @@
+// Copyright © 2023 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { Col, Row } from 'react-grid-system'
+import classNames from 'classnames'
+import { defineMessages } from 'react-intl'
+
+import Icon from '@ttn-lw/components/icon'
+import Button from '@ttn-lw/components/button'
+
+import Message from '@ttn-lw/lib/components/message'
+
+import attachPromise from '@ttn-lw/lib/store/actions/attach-promise'
+import PropTypes from '@ttn-lw/lib/prop-types'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import {
+  getUnseenNotifications,
+  updateNotificationStatus,
+} from '@console/store/actions/notifications'
+
+import {
+  selectNotifications,
+  selectTotalUnseenCount,
+  selectUnseenIds,
+} from '@console/store/selectors/notifications'
+import { selectUserId } from '@console/store/selectors/logout'
+
+import style from '../notifications.styl'
+
+import NotificationListItem from './list-item'
+
+const m = defineMessages({
+  archived: 'Archived notifications',
+  markAllAsRead: 'Mark all as read',
+})
+
+const NotificationList = ({
+  setSelectedNotification,
+  selectedNotification,
+  setShowContent,
+  fetchItems,
+  isArchive,
+  setShowListColumn,
+}) => {
+  const userId = useSelector(selectUserId)
+  const notifications = useSelector(selectNotifications)
+  const unseenIds = useSelector(selectUnseenIds)
+  const totalUnseenCount = useSelector(selectTotalUnseenCount)
+  const dispatch = useDispatch()
+  const isMobile = window.innerWidth < 768
+
+  const handleClick = useCallback(
+    async (e, id) => {
+      setShowListColumn(!isMobile)
+      setShowContent(true)
+      setSelectedNotification(notifications.find(notification => notification.id === id))
+      if (!isArchive) {
+        await dispatch(
+          attachPromise(updateNotificationStatus(userId, [id], 'NOTIFICATION_STATUS_SEEN')),
+        )
+        setTimeout(async () => {
+          await fetchItems()
+          await dispatch(attachPromise(getUnseenNotifications(userId)))
+        }, 300)
+      }
```

Comment:
Each notification selection now triggers three separate requests. As far as I can tell, you should really only need a single request, which is the one that updates the seen status **if** the notification was unseen before. If that request returns a 200 you can be sure enough that the message was marked correctly and set it as such in the update reducer.

What's the reason for refetching all notifications?

Also using arbitrary timeouts like here is only going to set you up for race conditions in slower environments.

Reply:
The reason for fetching all notifications was to update the store with the correct status and therefore update the visual, since the update request does not return the ids that were changed. But you are right, it is not necessary. Thanks for the suggestion. Haven't thought about this in that way.